### ISSUE TITLE
runtime error: slice bounds out of range [1:0]

### ISSUE THREAD

**Issue by vgjm** [dagger/dagger#1728]
I met this error when I run my plan by executing `dagger up` without europa.
The version is `dagger devel (0309c077) linux/amd64`.

The output is:
```
panic: runtime error: slice bounds out of range [1:0]

goroutine 2752 [running]:
go.dagger.io/dagger/cmd/dagger/logger.(*TTYOutput).printGroup(0xc00014e190, 0xc01083ed20, 0x79, 0xffffffffffffffff)
	/home/dagger/cmd/dagger/logger/tty.go:321 +0x60f
go.dagger.io/dagger/cmd/dagger/logger.(*TTYOutput).print(0xc00014e190)
	/home/dagger/cmd/dagger/logger/tty.go:203 +0x4a5
go.dagger.io/dagger/cmd/dagger/logger.(*TTYOutput).Write(0xc00014e190, {0xc009414800, 0x124, 0x1f4})
	/home/dagger/cmd/dagger/logger/tty.go:170 +0x187
github.com/rs/zerolog.levelWriterAdapter.WriteLevel(...)
	/root/go/pkg/mod/github.com/rs/zerolog@v1.26.1/writer.go:25
github.com/rs/zerolog.(*Event).write(0xc007ffaf00)
	/root/go/pkg/mod/github.com/rs/zerolog@v1.26.1/event.go:78 +0x126
github.com/rs/zerolog.(*Event).msg(0xc007ffaf00, {0x15360ee, 0x9})
	/root/go/pkg/mod/github.com/rs/zerolog@v1.26.1/event.go:142 +0x210
github.com/rs/zerolog.(*Event).Msg(...)
	/root/go/pkg/mod/github.com/rs/zerolog@v1.26.1/event.go:108
go.dagger.io/dagger/environment.(*Pipeline).Run(0xc00a0b4840, {0x1824a60, 0xc014f8ce10})
	/home/dagger/environment/pipeline.go:171 +0x53c
go.dagger.io/dagger/environment.(*Pipeline).Load(0xc00d3a0dc0, {0x1824a60, 0xc014f8ce10}, 0xc00a1ae180, {{0x0, 0x0}, 0xc01083b5e0, {0x12f5ba0, 0xc00d2ade70}, 0xc0091cacc0, ...})
	/home/dagger/environment/pipeline.go:708 +0x125
go.dagger.io/dagger/environment.(*Pipeline).doOp(0xc00d3a0dc0, {0x1824a60, 0xc014f8ce10}, 0x12be720, {{0x0, 0x0}, 0xc01083b5e0, {0x12f5ba0, 0xc00d2ade70}, 0xc0091cacc0, ...})
	/home/dagger/environment/pipeline.go:257 +0x318
go.dagger.io/dagger/environment.(*Pipeline).run(0xc00d3a0dc0, {0x1824a60, 0xc014f8ce10})
	/home/dagger/environment/pipeline.go:195 +0x170
go.dagger.io/dagger/environment.(*Pipeline).Run(0xc00d3a0dc0, {0x1824a60, 0xc014f8ce10})
	/home/dagger/environment/pipeline.go:146 +0x31c
go.dagger.io/dagger/environment.(*Pipeline).Load(0xc00d3a0b00, {0x1824a60, 0xc014f8ce10}, 0x30, {{0x0, 0x0}, 0xc01083a780, {0x12f5ba0, 0xc0078bfb90}, 0xc0091caa20, ...})
	/home/dagger/environment/pipeline.go:708 +0x125
go.dagger.io/dagger/environment.(*Pipeline).doOp(0xc00d3a0b00, {0x1824a60, 0xc014f8ce10}, 0x12be720, {{0x0, 0x0}, 0xc01083a780, {0x12f5ba0, 0xc0078bfb90}, 0xc0091caa20, ...})
	/home/dagger/environment/pipeline.go:257 +0x318
go.dagger.io/dagger/environment.(*Pipeline).run(0xc00d3a0b00, {0x1824a60, 0xc014f8ce10})
	/home/dagger/environment/pipeline.go:195 +0x170
go.dagger.io/dagger/environment.(*Pipeline).Run(0xc00d3a0b00, {0x1824a60, 0xc014f8ce10})
	/home/dagger/environment/pipeline.go:146 +0x31c
go.dagger.io/dagger/environment.newPipelineRunner.func1(0xc0009fc8c0)
	/home/dagger/environment/environment.go:145 +0x6e7
cuelang.org/go/tools/flow.RunnerFunc.Run(0x0, 0x0, {0x17f92a0, 0xc0000940f0})
	/root/go/pkg/mod/github.com/dagger/cue@v0.4.1-rc.1.0.20220121023213-66df011a52c2/tools/flow/flow.go:114 +0x1f
cuelang.org/go/tools/flow.(*Controller).runLoop.func1(0xc0009fc8c0)
	/root/go/pkg/mod/github.com/dagger/cue@v0.4.1-rc.1.0.20220121023213-66df011a52c2/tools/flow/run.go:67 +0x35
created by cuelang.org/go/tools/flow.(*Controller).runLoop
	/root/go/pkg/mod/github.com/dagger/cue@v0.4.1-rc.1.0.20220121023213-66df011a52c2/tools/flow/run.go:66 +0x31a
```

**Comment by shykes** [comment]
Hi @vgjm , were you able to resolve your issue?

Also, note that we recently released Dagger 0.2 with many improvements. If you need help migrating to the 0.2 API, we are happy to help!

**Comment by TomChv** [comment]
Has the same error with dagger europa, it comes from `maxLines` that is negative sometime (IDK why)

```
tty.go:336: Len(PrintEvents): 0; maxLines: -1
panic: runtime error: slice bounds out of range [1:0]

goroutine 169 [running]:
go.dagger.io/dagger/cmd/dagger/logger.(*TTYOutput).printGroup(0x14002bd1130, 0x14002ae6180, 0xa6, 0xffffffffffffffff)
        /Users/tomchauveau/Documents/DAGGER/dagger/cmd/dagger/logger/tty.go:338 +0x66c
go.dagger.io/dagger/cmd/dagger/logger.(*TTYOutput).print(0x14002bd1130)
        /Users/tomchauveau/Documents/DAGGER/dagger/cmd/dagger/logger/tty.go:214 +0x4f8
go.dagger.io/dagger/cmd/dagger/logger.(*TTYOutput).Write(0x14002bd1130, {0x14000dbe800, 0x90, 0x1f4})
        /Users/tomchauveau/Documents/DAGGER/dagger/cmd/dagger/logger/tty.go:181 +0x1e4
github.com/rs/zerolog.levelWriterAdapter.WriteLevel(...)
        /Users/tomchauveau/go/pkg/mod/github.com/rs/zerolog@v1.26.1/writer.go:25
github.com/rs/zerolog.(*Event).write(0x14012c7c0c0)
        /Users/tomchauveau/go/pkg/mod/github.com/rs/zerolog@v1.26.1/event.go:78 +0x170
github.com/rs/zerolog.(*Event).msg(0x14012c7c0c0, {0x10131a073, 0x9})
        /Users/tomchauveau/go/pkg/mod/github.com/rs/zerolog@v1.26.1/event.go:142 +0x1e8
github.com/rs/zerolog.(*Event).Msg(...)
        /Users/tomchauveau/go/pkg/mod/github.com/rs/zerolog@v1.26.1/event.go:108
go.dagger.io/dagger/plan.(*Runner).taskFunc.func1.1({{0x1018130c8, 0x14002bbd360}, 0x1, {0x0, 0x0}, {0x14000dbe600, 0x32, 0x1f4}, {0x14002bbd370, 0x1, ...}, ...})
        /Users/tomchauveau/Documents/DAGGER/dagger/plan/runner.go:163 +0xa4
go.dagger.io/dagger/plan.taskLog({0x1400222a0c0, 0x28}, 0x14012c7c060, {0x101801280, 0x1020e93b0}, 0x1017ed090)
        /Users/tomchauveau/Documents/DAGGER/dagger/plan/runner.go:130 +0x1f8
go.dagger.io/dagger/plan.(*Runner).taskFunc.func1(0x14002350d20)
        /Users/tomchauveau/Documents/DAGGER/dagger/plan/runner.go:162 +0x410
cuelang.org/go/tools/flow.RunnerFunc.Run(0x14004db7200, 0x14002350d20, {0x0, 0x0})
        /Users/tomchauveau/go/pkg/mod/github.com/dagger/cue@v0.4.1-rc.1.0.20220121023213-66df011a52c2/tools/flow/flow.go:114 +0x30
cuelang.org/go/tools/flow.(*Controller).runLoop.func1(0x14002350d20)
        /Users/tomchauveau/go/pkg/mod/github.com/dagger/cue@v0.4.1-rc.1.0.20220121023213-66df011a52c2/tools/flow/run.go:67 +0x3c
created by cuelang.org/go/tools/flow.(*Controller).runLoop
        /Users/tomchauveau/go/pkg/mod/github.com/dagger/cue@v0.4.1-rc.1.0.20220121023213-66df011a52c2/tools/flow/run.go:66 +0x2b0
```

As you see here 

```go
tty.go:336: Len(PrintEvents): 0; maxLines: -1
```

This simple if statement in the file [tty.go]() fix the issue

```go
		if len(printEvents) > maxLines && maxLines >= 0 {
			printEvents = printEvents[len(printEvents)-maxLines:]
		}
```
